tensorflow_serving/model_servers/http_server.cc

```diff
@@ -160,9 +160,20 @@ class RestApiRequestDispatcher {
     string output;
     VLOG(1) << "Processing HTTP request: " << req->http_method() << " "
             << req->uri_path() << " body: " << body.size() << " bytes.";
-    const auto status =
+    auto status =
         handler_->ProcessRequest(req->http_method(), req->uri_path(), body,
                                  &headers, &model_name, &method, &output);
+
+    if (req->http_method() == "OPTIONS") {
+      absl::string_view origin_header = req->GetRequestHeader("Origin");
+      if (RE2::PartialMatch(origin_header, "https?://")) {
+        status = Status::OK();
```

Comment:
return;

Reply:
I am not sure about this one.

Reply:
This needs to be done before handler_->ProcessingRequest() ... i.e .if the request is a CORS preflight, respond with OK and CORS headers directly; otherwise, proceed to handler_  ...   

If the request is an OPTIONS request but not with ORIGN header, respond with 404.